### ISSUE TITLE
getting best top N results

### ISSUE THREAD

**Issue by cguetot** [instadeepai/InstaNovo#12]
Hi, 

I read the InstaNovo paper on biorxiv. I followed the notebook code for making the beam search predictions. I wonder if it is possible to get the best top N results for each spectrum.
best, 

Carlos

**Comment by KevinEloff** [comment]
Hi Carlos,

Thank you for your interest. This functionality has been added in pull request #13, and is available in the latest release ([0.1.6](https://github.com/instadeepai/InstaNovo/releases/tag/0.1.6)). To return all beams, supply `return_all_beams=True` as an argument when calling [`decoder.decode`](https://github.com/instadeepai/InstaNovo/blob/b9a3b12eada59f83b44e356e37aebb9c81c42636/instanovo/inference/beam_search.py#L417).

For example, if you modified the [starter notebook](https://github.com/instadeepai/InstaNovo/blob/main/notebooks/getting_started_with_instanovo.ipynb):
```python
with torch.no_grad():
    p = decoder.decode(
        spectra=spectra,
        precursors=precursors,
        beam_size=config["n_beams"],
        max_length=config["max_length"],
        return_all_beams=True, # note: returns List[List[BeamState]] instead of List[BeamState]
    )
```
Now `p` will be returned as a list of [BeamStates](https://github.com/instadeepai/InstaNovo/blob/b9a3b12eada59f83b44e356e37aebb9c81c42636/instanovo/inference/beam_search.py#L18) sorted by model confidence for each batch item.

**Comment by cguetot** [comment]
Thanks
